### PR TITLE
upgrade: Make cluster health check at the start of services step 

### PR DESCRIPTION
**Pull request by jsuchome**
If the cluster is unhealthy, it could lead to errors during upgrade.
At this point all services are still running so it should be possible to
fix the problems.

I'm reusing existing health check functions here we are normally using for prechecks ... but why not?

### CODE REVIEW

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -416,10 +416,37 @@ def check_for_running_instances
...
+
+        msg = "Pacemaker cluster is not healthy. Fix the problem before repeating the upgrade step."
+        help_messages = []
+        clusters_health_report_errors(clusters_health).each do |problem, desc|
+          help_messages.push desc[:help]
+        end
+        Rails.logger.error msg
+        ::Crowbar::UpgradeStatus.new.end_step(
+          false,
+          services: {
```
Comment (skazi0) [inline crowbar_framework/app/models/api/upgrade.rb]: I'm not sure this is correct. There could be multiple errors from the cluster health check and we will loose the `data` fields for these while merging the `help` fields together. Can we merge the original structure with `services: {}` error? The clients should be able to handle this correctly.

Comment (jsuchome) [reply crowbar_framework/app/models/api/upgrade.rb]: Hmm... OK, do you find this better? (Ignore the fake data values)
```
| steps.services.status                                     | failed                                                                                                                                                                     |
| steps.services.errors.clusters_health_crm_failures.data   | crm_failures.values                                                                                                                                                        |
| steps.services.errors.clusters_health_crm_failures.help   | Please check that all cluster resources are online or refer to the SUSE High Availability documentation for possible troubleshooting.                                      |
| steps.services.errors.clusters_health_failed_actions.data | failed_actions.values,                                                                                                                                                     |
| steps.services.errors.clusters_health_failed_actions.help | Please clean the resource history and re−check the current state with "crm_resource -C" or refer to the SUSE High Availability documentation for possible troubleshooting. |
| steps.services.errors.clusters_health_unready_nodes.data  | unready_nodes.values                                                                                                                                                       |
| steps.services.errors.clusters_health_unready_nodes.help  | Some pacemaker nodes are in maintenance or standby node. Make sure all nodes are online and ready.  
```

Comment (jsuchome) [reply crowbar_framework/app/models/api/upgrade.rb]: Updated

**Comment by jsuchome** [comment]
services-cluster-health

@skazi0 do we need to update some labels in crowbar-ui?

**Comment by skazi0** [comment]
@jsuchome https://github.com/crowbar/crowbar-ui/pull/193

**Pull request merged**